tests/pytest/test_regression.py

```diff
@@ -35,45 +45,57 @@ def reference_dir(request):
 
 @pytest.fixture(params=config_filenames)
 def config(request):
+    print('Testing config', request.param)
     config_filename = os.path.join(CONFIG_DIR, request.param)
     with open(config_filename, 'r') as config_file:
         return yaml.safe_load(config_file)
 
 
-def test_regression(config, model_image, reference_dir):
+def test_regression(config, model_image, model_image_tag, code_root, reference_dir):
     run_name = config['experiment_name']
     run_reference_dir = os.path.join(reference_dir, run_name)
     run_dir = os.path.join(OUTPUT_DIR, run_name)
     if os.path.isdir(run_dir):
         shutil.rmtree(run_dir)
     os.makedirs(run_dir)
     write_run_directory(config, run_dir)
-    run_model(run_dir, MODEL_IMAGE)
+    run_model(run_dir, model_image, model_image_tag, code_root)
     md5sum_filename = os.path.join(run_reference_dir, MD5SUM_FILENAME)
     if not os.path.isfile(md5sum_filename):
-        assert False, (f"reference md5sum does not exist, "
+        assert False, (f"reference md5sum does not exist at " + md5sum_filename + ","
                        f"you can create it with `bash set_reference.sh {reference_dir}`"
                        " after running the model and generating the output directories."
                        )
     else:
+        print('Checking',  run_dir, 'and',  md5sum_filename)
         check_md5sum(run_dir, md5sum_filename)
+        if model_image_tag == 'serialize':
+            nonserialized_comparison(run_dir, run_reference_dir)
     shutil.rmtree(run_dir)
 
+def nonserialized_comparison(run_dir, run_reference_dir):
+    non_serialized_refdir = run_reference_dir.replace('_serialize', '')
+    md5sum_filename = os.path.join(non_serialized_refdir, MD5SUM_FILENAME)
+    if not os.path.isfile(md5sum_filename):
+        assert False, (f"equivalent non-serialized reference md5sum does not exist at " + md5sum_filename)
+    print('Checking netcdf files match non-serialized equivalent', md5sum_filename)
+    check_md5sum(run_dir, md5sum_filename)
```

Comment:
You can remove this, since the tests already check that the non-serialized and serialized image both match the same reference md5sum.